data/src/main/java/com/github/abdallahabdelfattah13/data/DataInjection.kt

```diff
@@ -11,6 +17,26 @@ import com.github.abdallahabdelfattah13.domain.repositories.feed.FeedRepository
  */
 object DataInjection {
 
+    private lateinit var mFeedDatabase: FeedDatabase
+
+    @JvmStatic
+    fun provideFeedInMemory(): InMemoryFeedRepo = InMemoryFeedRepoImpl()
+
+    @JvmStatic
+    fun provideDatabase(context: Context): FeedDatabase {
+        mFeedDatabase = Room.databaseBuilder(
+            context,
+            FeedDatabase::class.java,
+            "feed.db"
+        )
+            .build()
+        return mFeedDatabase
+    }
+
+    @JvmStatic
+    fun provideDatasource(): DataSourceFactory =
+        DataSourceFactory(provideFeedInMemory(), mFeedDatabase)
+
     @JvmStatic
-    fun provideFeedRepository(): FeedRepository = InMemoryFeedRepository()
+    fun provideRepo(): FeedRepository = FeedRepositoryImpl(provideDatasource())
```

Comment:
I think we should keep the old name. We can have more repositories in the future.

Reply:
Ok